next.config.js

```diff
@@ -14,4 +16,20 @@ module.exports = {
       },
     ];
   },
+  webpack: (config) => {
+    config.resolve.alias = {
+      ...(config.resolve.alias || {}),
+      // The following aliases are added to ensure that both the Next.js instance and
+      // the `veda-ui` library use the same instances of Jotai for state management.
+      // This resolves the issue of "Detected multiple Jotai instances," which can cause
+      // unexpected behavior due to multiple instances of Jotai's default store.
+      // For more details, refer to the GitHub discussion:
+      // https://github.com/pmndrs/jotai/discussions/2044
+      'jotai': path.resolve(__dirname, 'node_modules', 'jotai'),
+      'jotai-devtools': path.resolve(__dirname, 'node_modules', 'jotai-devtools'),
+      'jotai-location': path.resolve(__dirname, 'node_modules', 'jotai-location'),
+      'jotai-optics': path.resolve(__dirname, 'node_modules', 'jotai-optics'),
```

Comment:
Looks like yarn/npm seem to hoist `node_modules` to the top directory, if they are referenced within the project. I initially thought I'd have to write `node_modules/@developmentseed/veda-ui/node_modules/jotai`, but this seems to work as well 🤔

Reply:
Aha, 
this makes me wonder if it will be better just to export these hooks that use atoms from veda-ui side, so the client doesn't even need to install jotai separately. not a blocker, I will experiment this while resolving the modal issue.

Reply:
Does that mean that instances would need to install jotai as a peer dependency, if we don't ship it via the UI library? I also tried using the `<Provider />` mentioned [here](https://jotai.org/docs/guides/nextjs), but couldn't get rid of the "multiple instances" problem. Might be worthwhile to re-visit again.

Reply:
🤔 My current thought is that instances don't even need jotai as a peer dependency, (like jotai will be a dependency of VEDA UI not the instance). I haven't tried it yet! let me try and revisit this again!